src/scaling.jl

```diff
@@ -0,0 +1,189 @@
+"""
+    scale(f=std, x, y=f(skipmissing(x)))
+
+Scale an array `x` by a scalar `y`.
+
+
+!!! warning
+    This only scales and does not center the values, unlike `scale` in R.
+    See `StatsBase.zscore` for that functionality.
+
+See also [`scale!`](@ref)
+"""
+scale(x) = scale(std, x)
+scale(f::Function, x) = scale(x, f(skipmissing(x)))
+scale(x, y) = x ./ y
+
+"""
+    scale(f=std, x, y=f(skipmissing(x)))
+
+Scale an array `x` in place by a scalar `y`.
+
+!!! warning
+    This only scales and does not center the values, unlike `scale` in R.
+    See `StatsBase.zscore` for that functionality.
+
+See also [`scale`](@ref)
+"""
+scale!(x) = scale!(std, x)
+scale!(f::Function, x) = scale!(x, f(skipmissing(x)))
+
+function scale!(x, y)
+    try
+        y = convert(eltype(x), y)
+        # verify that we also don't hit problems during substraction
+        # I can't think of any common number type that isn't closed under
+        # subtraction, but maybe somebody has created a PositiveInt type
+        convert(eltype(x), first(x) / y)
+    catch e
+        if e isa InexactError
+            throw(ArgumentError("Dividing the scale $(y) changes the eltype of " *
+                                "the array. Promote to $(typeof(first(x) - y)) first."))
+        else
+            rethrow(e)
+        end
+    end
```

Comment:
I would just do `x ./= y` and let that give an appropriate error rather than trying to catch the error ahead of time, i.e.
```suggestion
```
If you do want to catch it first, I'd do it this way:
```suggestion
    if !(typeof(one(eltype(x)) / one(typeof(y))) <: eltype(x))
        throw(ArgumentError("Dividing the scale $y changes the eltype of " *
                            "the array. Promote to $(eltype(x)) first."))
    end
```

Reply:
I was hoping to give something nicer than a big opaque stacktrace (because this can happen deep in other code) and  I can imagine "non programmers" using this as just another stats tool. The other thing was trying for consistency with `center!` which does do this check. BUT for `center!` this is informative because it's subtraction and you'll generally do fine for all elements if you can coerce the center to Int. But for division ... you might success for some elements but not for all depending on whether the scale divides a particular element.

Reply:
(Also why the test via `one()` won't work because for 1/1 is always 1 in a ring.)

Reply:
It will be 1 of the appropriate type, that's all this doing